### PR TITLE
feat: add TikTok author_collections endpoint

### LINKED ISSUE #740: A new API endpoint for accessing detailed information about user collections, such as their names.
About it's scope: It would also add a new collectionId parameter to fetch_user_collect. When specified, this parameter allows you to fetch information about a specific collection. I already worked on a PR in v4 and would migrate it to v5, if I'm allowed to.

### PR DESCRIPTION
**Pull request by BennoCrafter**
## Why / 为什么
Currently it's possible to fetch all saved posts from a TikTok user or all saved posts from a specific collection using its ID. However, there is currently no way to find a collection's ID without manually inspecting the browser.
The new endpoint I added allows users to get detailed information about all of their collections, including their IDs. (resolves #740)

Example response:
```json
{
  "name": "Wanna go to",
      "cover": {
        "url": ",
        "urls": [
          "",
          "",
          ""
        ],
        "width": null,
        "height": null
      },
  "platform": "tiktok",
  "item_count": 49,
  "collection_id": "7659011279817935649"
},
```

or in raw format:
```json
{
    "name": "Wanna go to",
    "cover": {
      "urlList": [
        "url",
        "url",
        "url"
      ]
    },
    "total": "49",
    "status": 1,
    "userId": "userId",
    "userName": "userName",
    "collectionId": "7659011279817935649"
},
```

I'm don't know, if it's possible to implement this for Douyin too, just because I don't know nor use it.

## Checklist / 自查

- [ x ] `make fmt` — 格式化先跑，免得 CI 挂在空白字符上 / run first, so the gate does not fail on whitespace
- [ x ] `make lint && make type` — ruff and mypy
- [ x ] `make test` — unit, replay and integration
- [ x ] `(cd web && npm run verify)` — 动过 `web/` 才需要 / only if you touched `web/`
- [ x ] 文档 `documents/en/` **和** `documents/zh/` 一起更新了 / user documentation updated in **both**, if a reader would act on the change (a setting, a CLI command, an endpoint, a console page, a default)
- [ x ] 没有：源码里的中日韩字符、tokens 之外的十六进制颜色、JSX 里的字面文案、`TODO`、`.env`、带真实 Cookie 的 fixture / none of: CJK in source, a hex colour outside the tokens file, literal prose in JSX, a `TODO`, a `.env`, a fixture with a real cookie
- [ x ] 没有在 diff 或描述里贴出密钥、口令、Cookie 或 API Key / no key, password, cookie or API key in the diff or in this description

### CODE REVIEW

File: src/dtk/i18n/locales/zh.json [hunk 1]
```diff
@@ -254,6 +254,10 @@
         "summary": "登出其他所有设备",
         "description": "退出除当前设备之外的所有登录。"
       },
+      "author_collections": {
```
Comment (BennoCrafter) [inline src/dtk/i18n/locales/zh.json:257]: check translation

File: web/src/locales/zh/console.json [hunk 2]
```diff
@@ -1167,6 +1167,7 @@
       "user": "作者资料",
       "posts": "作者作品列表",
       "likes": "作者点赞列表",
+      "collections": "作者收藏夹列表",
```
Comment (BennoCrafter) [inline web/src/locales/zh/console.json:1170]: check translation

**Comment by BennoCrafter** [comment]
Another suggestion would be to separate the “Mix/Playlist” endpoint from collection fetching. Imo, it’s not clear that this endpoint can also be used to fetch a user’s TikTok collections.

**Comment by Evil0ctal** [comment]
Reviewed this properly rather than skimming it — pulled the branch, merged it onto current `main` in a worktree and ran everything. Thank you for following the conventions in #740 closely; most of what I asked for is here and in the right places.

**What passed.** Merges onto `main` with no conflicts even though the branch is 27 commits behind it. `ruff check`, `ruff format --check` and `mypy` are clean. 2474 unit and replay tests pass, 505 integration tests pass, and the console's `typecheck`, `eslint` and `check-i18n.mjs` all pass — the i18n is in both catalogues in both languages and the Chinese is written rather than machine-translated, which I appreciate.

The structural work is right: `risk_weight=1.8` matching the other list endpoints, a builder that starts from `base_params(profile)` and leaves values un-encoded, strict parsing through the `common.py` helpers, an `EndpointPolicy`, the registry triple (`Capability`, `_ARGUMENTS`, `_CACHE_TTL_KEYS`), the route copied from `mix_posts`, nothing added to MCP, and nothing added to `P0_CAPABILITIES`. The `DouyinAdapter.parse_author_collections` stub raising `UnsupportedContent` is exactly the right call.

---

## One real bug: `cover` is always `null`

Your own PR description shows it. The raw capture has

```json
"cover": { "urlList": ["url", "url", "url"] }
```

and the normalized output right above it has

```json
"cover": null
```

`image_from_urls` takes bare URL strings or sequences of them ([`common.py:188`](https://github.com/Evil0ctal/Douyin_TikTok_Download_API/blob/main/src/dtk/platforms/common.py#L188-L204)). A `Mapping` is neither, so it falls through to an empty list and returns `None`. `image_from_url_list` is the container-shaped one, but it reads `url_list` — the Douyin/aweme spelling — not TikTok's `urlList`, so it does not help here either.

I ran your own payload through it to be sure:

```
collection_id  7659011279817935649
name           Wanna go to
item_count     49            <- fine, optional_int handles the string "49"
cover          None          <- upstream sent two URLs
```

The fix is to hand it the list rather than the container:

```python
cover=image_from_urls((node.get("cover") or {}).get("urlList")),
```

## Why the suite did not catch it

`_COLLECTIONS_PAGE1` uses `"cover": "https://…"` — a bare string. TikTok does not send that shape; your own capture says so. The test passes because the payload was written to match the parser instead of the platform, so the assertion `first.cover.url == …` proves only that the helper works on a string nobody sends.

This is the part I care most about, and it is the reason #740 asked for a capture. Please replace the inline literal with the shape from your description — `cover` as `{"urlList": [...]}`, `total` as the string `"49"` — and keep the `status`/`userId`/`userName` keys so the fixture is the real envelope rather than a subset of it. Redact the user fields.

Your note in `endpoints.py` says the envelope is "inferred by analogy … must be confirmed against a live capture before release". You clearly have that capture — it is in the PR description. Folding it into the test settles that comment too.

## Smaller things

**The parser docstring contradicts the endpoint spec.** `parse_author_collections` says "see `empty_body_is_normal` on the endpoint spec", and the spec deliberately does not set it and spends a paragraph explaining why. The spec's reasoning is right; the docstring pointer should go.

**The branch is 27 commits behind `main`.** It merges cleanly, but `main` has moved through 5.0.1 → 5.0.3 and touched `models/content.py` and the TikTok parser. Please merge `main` in so CI runs against what will actually land.

**Scope.** #740 asked for Douyin's `collects/list/`; this is TikTok's `collection_list/`. It is worth having on its own and I am happy to take it, but it does not resolve that issue, so I will drop the `resolves #740` line when merging and leave #740 open for the Douyin half.

---

Fix the `cover` mapping, put the real shape in the test, drop the stale docstring line, and merge `main` in — then I will merge this. I have approved the workflow run so CI reports on the next push.

On renaming mix/playlist: you are right that it is unclear. That is a separate change and I will open an issue for it rather than have it ride along here.

**Comment by BennoCrafter** [comment]
It's also worth noting that when I tested this with real cookies, I found that the maximum number retrieved is not 50—as hard-coded in the translation—but 35. I didn't look into this issue further because I wasn't sure how to fix it. The API fails for any value above this limit.

**Comment by Evil0ctal** [comment]
Merged. Thank you — that was a clean turnaround.

I checked the fix rather than taking it on trust: ran your own capture's shape through the parser and got both cover URLs and `item_count: 49` from the string `"49"`. The replay payload is the real envelope now, redacted, and updating the `endpoints.py` comment from "inferred by analogy … must be confirmed against a live capture" to what the capture actually showed is exactly the right move — that comment is the thing a future reader will trust.

Squashed rather than merged as three, and I rewrote the commit body to record why `cover` was `null` and why the test did not catch it. That failure mode — a test written to match the parser instead of the platform — is worth having in the log.

One thing I could not prevent: GitHub closed #740 on merge because the PR description says `resolves #740`. I have reopened it. This landed the TikTok half; the Douyin `collects/list/` half is still unwired and that issue tracks it.

Your two other findings are both real and both now tracked:

- **#755** — the page-size ceiling. `MAX_PAGE_SIZE = 50` is shared by every list endpoint and published as `le=50` on the parameter, so if TikTok refuses above ~35 we are advertising a size that does not exist, and a caller who tries it spends identity health to find out. I need the failing response before setting a cap; the issue says exactly what would settle it.
- **#754** — you were right and I was wrong about the framing. I opened it as a label change; if `mix_posts` really does double as collection fetching then a separate endpoint is the fix, not better wording. I have written up what the tree says there, including the gap your PR made visible: there is now no endpoint that consumes a `collectionId`. The question I cannot answer without your cookies is which path you used and where you put the id.

Both are yours if you want them.
